client/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -0,0 +1,50 @@
+package org.http4s
+package client
+package middleware
+
+import org.http4s.Status.ResponseClass.Successful
+import org.http4s.{Response, Request, EmptyBody}
+import org.http4s.client.Client
+import scala.concurrent.duration._
+import scala.math.{pow, min, random}
+import scalaz.concurrent.Task
+import scala.language.postfixOps
+
+object Retry {
+
+  def apply(backoff: Task[Int => Option[FiniteDuration]])(client: Client) = new Client {
+
+    def shutdown(): Task[Unit] = client.shutdown()
+
+    def prepare(req: Request): Task[Response] = prepareLoop(req, 1)
+
+    private def prepareLoop(req: Request, attempts: Int): Task[Response] = {
+      client.prepare(req) flatMap {
+        case Successful(resp) => Task.now(resp)
+        case fail => backoff flatMap { f =>
```

Comment:
Agreed, a log message at the minimum, but an accumulation of failures with type would be helpful.